classes/PHPTAL/Php/Attribute/TAL/Attributes.php

```diff
@@ -46,6 +46,8 @@ class PHPTAL_Php_Attribute_TAL_Attributes
      * value for default keyword
      */
     private $_default_escaped;
+    private string $_attribute;
```

Comment:
I intentionally left it uninitialized: if tests pass, it means that initialization always happens before consume, and we are safe to discard the nullable type.

We can add the nullable type back if and when someone opens an issue about it with real life examples.

If it sounds like _testing in production_ it's because that's exactly it :wink:

Reply:
> if tests pass, it means that initialization always happens before consume

Our tests, and this is a very old codebase :P 

Practically, PHPStan/Psalm would've been annoying you for a while here, but this codebase is not covered by any kind of static analysis: this isn't sound at type level.

Reply:
@Slamdunk this is the only point left to solve BTW: happily releasing afterwards

Reply:
This has already been updated, isn't it?

Reply:
Ah yes, thanks!